src/components/chatbot/ChatBox.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useEffect, useState } from 'react'
+import ChatBoxForm from './ChatBoxForm';
+import ChatMessage from './ChatMessage';
+import { useChat } from '@/hooks/useChat';
+import { useSendMessage } from '@/hooks/useBot';
+import { ByUser, Message } from '@/types/messages/message';
+import { getMessagesFromChat } from '@/lib/chat';
+
+type P = {
+  // ID of the chat to retrieve history of messages from. Not the TG Chat ID, but the actual chat Id stored in database
+  chatId: number;
+}
+
+export default function ChatBox({chatId}: P) {
+  const {data: chat, isPending} = useChat(chatId);
+  const sendMessageMutation = useSendMessage(chatId)
+
+  const [messages, setMessages] = useState<Message[]>([])
+  
+  useEffect(() => {
+    if (chat) {
+      setMessages(getMessagesFromChat(chat));
+    }
+
+  }, [chat])
+  
+  useEffect(() => {
+    try {
+      // TODO: Replace URL with constant string
+      const socket = new WebSocket('ws://localhost:8080/api/ws');
+  
+      socket.onopen = () => {
+        console.log('WebSocket connection established');
+      };
+  
+      socket.onmessage = (event) => {
+        setMessages((prevMessages) => [...prevMessages, event.data])
```

Comment:
```suggestion
        setMessages((prevMessages) => [
          ...prevMessages,
          JSON.parse(event.data).data,
        ]);
```

there was an issue with event.data being a string and it not parsing correctly 🤔

Reply:
Strange that it actually still worked on my end, but yeah, will fix